src/OrleansProviders/Streams/Generator/GeneratorPooledCache.cs

```diff
@@ -58,13 +60,65 @@ public bool Equals(CachedMessage cachedMessage, IStreamIdentity streamIdentity)
             }
         }
 
+        private class ExplicitEvictionStrategy : IEvictionStrategy<CachedMessage>
```

Comment:
Ah,  I see.  We can look at reducing code duplication later if/when needed.  Shouldn't block this PR.